### PR TITLE
EZR: Use VANotify to send submission failure email to Veteran

### PR DESCRIPTION
**Pull request by dellerbie**
## Summary

- *This work is behind a feature toggle (flipper): YES `ezr_use_va_notify_on_submission_failure`*
- Use VANotify to send a submission failure email to the Veteran. This work is part of the Zero Silent Failures initiative
- 1010 Health Apps Team
- The email will only be sent when the Flipper is enabled

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#92911
- https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/3112

## Testing done

- [x] *New code is covered by unit tests*
QA will be performed in Staging and behind the Flipper.

## Screenshots
coming

## What areas of the site does it impact?
Form 10-10 EZR

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/sidekiq/hca/ezr_submission_job.rb [hunk 1]
```diff
@@ -38,6 +40,22 @@ def self.decrypt_form(encrypted_form)
...
+      salutation = first_name ? "Dear #{first_name}," : ''
+
+      VANotify::EmailJob.perform_async(
+        email,
+        template_id,
+        { 'salutation' => salutation },
+        api_key
+      )
+      StatsD.increment("#{STATSD_KEY_PREFIX}.submission_failure_email_sent")
+    end
```
[the remark this answers is not in the document]
Comment (coope93) [reply app/sidekiq/hca/ezr_submission_job.rb:57]: I think if we are going to use different templates it makes sense to keep it separate. If we decide to use the same template maybe we consider refactoring it to be reusable, but I think it's fine for now.

File: config/settings.yml [hunk 2]
```diff
@@ -1386,6 +1386,7 @@ vanotify:
       api_key: fake_secret
       template_id:
         form1010_ez_failure_email: fake_template_id
+        form1010_ezr_failure_email: fake_template_id
```
Comment (coope93) [inline config/settings.yml:1389]: Just wanted to call out that if you haven't created it yet you'll need to update the `vsp-infra-application-manifests` repo similar to [this PR](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2923).

**Review by coope93 — DISMISSED** [review]

**Review by JoshingYou1 — APPROVED** [review]
LGTM! Nice work Derrick 😄

File: app/sidekiq/hca/ezr_submission_job.rb [hunk 3]
```diff
@@ -38,6 +40,22 @@ def self.decrypt_form(encrypted_form)
       JSON.parse(HealthCareApplication::LOCKBOX.decrypt(encrypted_form))
     end
 
+    def self.send_failure_email(parsed_form)
+      email = parsed_form['email']
```
Comment (coope93) [inline app/sidekiq/hca/ezr_submission_job.rb:44]: Do we always have an email here? I know on the 10-10EZ we have already validated if the email exists by this time. I just wanted to confirm that we know the form has an email.

**Review by ryan-mcneil — APPROVED** [review]

**Review by coope93 — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]

**Review by JoshingYou1 — APPROVED** [review]

**Pull request merged**